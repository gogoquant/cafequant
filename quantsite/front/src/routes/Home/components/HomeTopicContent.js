import React from 'react';
import PropTypes from 'prop-types';
import { Divider } from 'antd';
import { timeUtils, regexUtils } from 'utils';
import Ellipsis from 'components/Ellipsis';
import styles from './HomeTopicContent.scss';

const HomeTopicContent = ({ topic }) => (
  <div>
    <Ellipsis text={topic.content.replace(regexUtils.htmlElementRegex, '')} />
    <div className={styles.info}>
      {`阅读数:${topic.visit_count}`}
      <Divider type="vertical" />
      {`${topic.reply_count}条评论`}
      <Divider type="vertical" />
      {timeUtils.fromNow(topic.last_reply_at)}
    </div>
  </div>
);

HomeTopicContent.propTypes = {
  topic: PropTypes.object.isRequired,
};

export default HomeTopicContent;
