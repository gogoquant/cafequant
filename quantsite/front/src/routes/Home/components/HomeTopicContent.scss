.info {
  margin-top: 16px;
}
