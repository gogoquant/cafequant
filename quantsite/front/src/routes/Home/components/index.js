export { default as HomeTopics } from './HomeTopics';
