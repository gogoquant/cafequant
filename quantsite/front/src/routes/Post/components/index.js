export { default as PostTab } from './PostTab';
export { default as PostTitle } from './PostTitle';
