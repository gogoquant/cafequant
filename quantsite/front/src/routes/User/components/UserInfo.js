import React from 'react';
import PropTypes from 'prop-types';
import { timeUtils } from 'utils';
import { Card, Avatar, Tag } from 'antd';
import styles from './UserInfo.scss';

const UserInfo = ({ loading, user }) => (
  <Card className={styles.container} loading={loading}>
    <Card.Meta
      avatar={<Avatar src={user.avatar_url} />}
      title={<Tag color="purple">{user.loginname}</Tag>}
      description={
        <React.Fragment>
          <div>
            Github：
            <a
              target="_blank"
              href={`https://github.com/${user.githubUsername}`}
              rel="noopener noreferrer"
            >
              {user.githubUsername}
            </a>
          </div>
          <div>积分：{user.score}</div>
          <div>收藏话题：{user.collect && user.collect.length} 个</div>
          <div>注册时间：{timeUtils.fromNow(user.create_at)}</div>
        </React.Fragment>
      }
    />
  </Card>
);

UserInfo.propTypes = {
  loading: PropTypes.bool.isRequired,
  user: PropTypes.object.isRequired,
};

export default UserInfo;
