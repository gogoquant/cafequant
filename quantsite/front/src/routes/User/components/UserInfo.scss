.container {
  width: 100%;
  margin-bottom: 24px;
}
