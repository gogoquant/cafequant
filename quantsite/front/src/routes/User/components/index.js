export { default as UserInfo } from './UserInfo';
export { default as UserTopics } from './UserTopics';
