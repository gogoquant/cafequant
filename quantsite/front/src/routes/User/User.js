import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import classNames from 'classnames';
import Title from 'components/Title';
import Exception from 'components/Exception';
import { getUserData } from './UserRedux';
import { UserInfo, UserTopics } from './components';
import styles from './User.scss';

const cls = classNames(styles.mdContainer);

export class UserComponent extends React.Component {
  constructor(props) {
    super(props);
    this.getUser = this.getUser.bind(this);
  }

  componentDidMount() {
    this.getUser();
  }

  componentDidUpdate(prevProps) {
    const {
      match: {
        params: { name: prevName },
      },
    } = prevProps;
    const {
      match: {
        params: { name },
      },
    } = this.props;
    if (name !== prevName) {
      this.getUser();
    }
  }

  getUser() {
    const {
      match: {
        params: { name },
      },
    } = this.props;
    this.props.getUserData(name);
  }

  render() {
    const { loading, userData, error } = this.props;
    const { loginname: username } = userData;
    const title = username === undefined ? '用户中心' : `@${username}`;

    if (error) {
      return <Exception />;
    }
    return (
      <div className={cls}>
        <Title title={title} />
        <UserInfo loading={loading} user={userData} />
        {!loading && (
          <UserTopics
            dataList={{
              topics: userData.recent_topics,
              replies: userData.recent_replies,
              collect: userData.collect,
            }}
          />
        )}
      </div>
    );
  }
}

UserComponent.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      name: PropTypes.string,
    }),
  }).isRequired,

  loading: PropTypes.bool,
  userData: PropTypes.object,
  error: PropTypes.bool,
  getUserData: PropTypes.func,
};

export default connect(
  state => ({
    loading: state.getIn(['user', 'loading']),
    userData: state.getIn(['user', 'userData']),
    error: state.getIn(['user', 'error']),
  }),
  {
    getUserData,
  },
)(UserComponent);
