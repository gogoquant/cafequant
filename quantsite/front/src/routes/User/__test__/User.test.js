import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Map } from 'immutable';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import MockDate from 'mockdate';
import configureStore from 'redux-mock-store';
import User, { UserComponent } from '../User';
import { UserTopics } from '../components';

MockDate.set(new Date('2018-05-21T16:00:00.000Z'));

describe('<User />', () => {
  const userProps = {
    loading: false,
    userData: {
      avatar_url: '//',
      collect: [{}, {}],
      create_at: '2018-05-20T16:00:00.000Z',
      githubUsername: 'admin',
      loginname: 'admin',
      score: 520,
      recent_topics: [
        {
          author: {
            avatar_url: '//',
            loginname: 'admin',
          },
          id: '007',
          title: 'title',
          last_reply_at: '2018-05-20T16:00:00.000Z',
        },
      ],
      recent_replies: [],
    },
    error: false,
  };

  const middleware = [thunk];
  const mockStore = configureStore(middleware);

  const userInitialState = Map({
    user: Map(userProps),
    getUserData: jest.fn(),
  });

  const routeProps = {
    match: {
      params: {
        name: 'admin',
      },
    },
  };

  it('should render correctly', () => {
    const store = mockStore(userInitialState);
    const tree = mount(
      <MemoryRouter>
        <User store={store} {...routeProps} />
      </MemoryRouter>,
    );
    expect(toJson(tree, { noKey: true, mode: 'deep' })).toMatchSnapshot();
  });

  it('should render default title', () => {
    const userData = {
      ...userProps.userData,
      loginname: undefined,
    };
    const store = mockStore(
      userInitialState.setIn(['user', 'userData'], userData),
    );
    const tree = mount(
      <MemoryRouter>
        <User store={store} {...routeProps} />
      </MemoryRouter>,
    );
    expect(toJson(tree, { noKey: true, mode: 'deep' })).toMatchSnapshot();
  });

  it('should render <Exception />', () => {
    const initialState = userInitialState.setIn(['user', 'error'], true);
    const store = mockStore(initialState);
    const tree = mount(
      <MemoryRouter>
        <User store={store} {...routeProps} />
      </MemoryRouter>,
    );
    expect(toJson(tree, { noKey: true, mode: 'deep' })).toMatchSnapshot();
  });

  it('should call getUser() when name change and not when name same', () => {
    const props = {
      ...userProps,
      getUserData: jest.fn(),
    };
    const spy = jest.spyOn(UserComponent.prototype, 'getUser');
    const wrapper = shallow(<UserComponent {...routeProps} {...props} />);
    wrapper.setProps({
      match: {
        params: {
          name: 'guest',
        },
      },
    });
    expect(spy).toHaveBeenCalledTimes(2);

    wrapper.setProps({
      match: {
        params: {
          name: 'guest',
        },
      },
    });
    expect(spy).toHaveBeenCalledTimes(2);
  });
});

describe('<UserTopics />', () => {
  it('should change state when call onTableChange()', () => {
    const dataListMock = {
      collect: [],
      replies: [],
      topics: [],
    };

    const wrapper = shallow(<UserTopics dataList={dataListMock} />);
    wrapper.instance().onTabChange('topics');
    expect(wrapper.state().key).toEqual('topics');
  });
});
