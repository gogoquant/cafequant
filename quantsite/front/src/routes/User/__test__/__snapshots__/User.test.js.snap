// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should render <Exception /> 1`] = `
<ConnectFunction
  match={
    Object {
      "params": Object {
        "name": "admin",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <span>
      网络出小差了！请稍后再试
    </span>
  </div>
</ConnectFunction>
`;

exports[`<User /> should render correctly 1`] = `
<ConnectFunction
  match={
    Object {
      "params": Object {
        "name": "admin",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className=""
  >
    <div
      className="ant-card ant-card-bordered"
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-avatar"
          >
            <span
              className="ant-avatar ant-avatar-circle ant-avatar-image"
              style={Object {}}
            >
              <img
                onError={[Function]}
                src="//"
              />
            </span>
          </div>
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              <div
                className="ant-tag ant-tag-purple"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              >
                admin
              </div>
            </div>
            <div
              className="ant-card-meta-description"
            >
              <div>
                Github：
                <a
                  href="https://github.com/admin"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  admin
                </a>
              </div>
              <div>
                积分：
                520
              </div>
              <div>
                收藏话题：
                2
                 个
              </div>
              <div>
                注册时间：
                a day ago
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card ant-card-bordered ant-card-contain-tabs"
    >
      <div
        className="ant-card-head"
        style={Object {}}
      >
        <div
          className="ant-card-head-wrapper"
        />
        <div
          className="ant-tabs ant-tabs-top ant-card-head-tabs ant-tabs-large ant-tabs-line"
          onScroll={[Function]}
          style={Object {}}
        >
          <div
            className="ant-tabs-bar ant-tabs-top-bar ant-tabs-large-bar"
            onKeyDown={[Function]}
            role="tablist"
            style={Object {}}
            tabIndex="0"
          >
            <div
              className="ant-tabs-nav-container"
            >
              <span
                className="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
                onClick={null}
                onTransitionEnd={[Function]}
                unselectable="unselectable"
              >
                <span
                  className="ant-tabs-tab-prev-icon"
                >
                  <i
                    aria-label="icon: left"
                    className="anticon anticon-left ant-tabs-tab-prev-icon-target"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </i>
                </span>
              </span>
              <span
                className="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
                onClick={null}
                unselectable="unselectable"
              >
                <span
                  className="ant-tabs-tab-next-icon"
                >
                  <i
                    aria-label="icon: right"
                    className="anticon anticon-right ant-tabs-tab-next-icon-target"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                      />
                    </svg>
                  </i>
                </span>
              </span>
              <div
                className="ant-tabs-nav-wrap"
              >
                <div
                  className="ant-tabs-nav-scroll"
                >
                  <div
                    className="ant-tabs-nav ant-tabs-nav-animated"
                  >
                    <div>
                      <div
                        aria-disabled="false"
                        aria-selected="true"
                        className="ant-tabs-tab-active ant-tabs-tab"
                        onClick={[Function]}
                        role="tab"
                        style={
                          Object {
                            "marginRight": null,
                          }
                        }
                      >
                        最近创建的话题
                      </div>
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        className=" ant-tabs-tab"
                        onClick={[Function]}
                        role="tab"
                        style={
                          Object {
                            "marginRight": null,
                          }
                        }
                      >
                        最近参与的话题
                      </div>
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        className=" ant-tabs-tab"
                        onClick={[Function]}
                        role="tab"
                        style={
                          Object {
                            "marginRight": null,
                          }
                        }
                      >
                        收藏的话题
                      </div>
                    </div>
                    <div
                      className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            onKeyDown={[Function]}
            role="presentation"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
                "position": "absolute",
                "width": 0,
              }
            }
            tabIndex={0}
          />
          <div
            className="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
            style={
              Object {
                "marginLeft": "0%",
              }
            }
          >
            <div
              aria-hidden="false"
              className="ant-tabs-tabpane ant-tabs-tabpane-active"
              role="tabpanel"
            >
              <div
                onKeyDown={[Function]}
                role="presentation"
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": 0,
                  }
                }
                tabIndex={0}
              />
              <div
                onKeyDown={[Function]}
                role="presentation"
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": 0,
                  }
                }
                tabIndex={0}
              />
            </div>
            <div
              aria-hidden="true"
              className="ant-tabs-tabpane ant-tabs-tabpane-inactive"
              role="tabpanel"
            />
            <div
              aria-hidden="true"
              className="ant-tabs-tabpane ant-tabs-tabpane-inactive"
              role="tabpanel"
            />
          </div>
          <div
            onKeyDown={[Function]}
            role="presentation"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
                "position": "absolute",
                "width": 0,
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-list ant-list-split"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div
              className="ant-spin-container"
            >
              <ul
                className="ant-list-items"
              >
                <li
                  className="ant-list-item"
                >
                  <div
                    className="ant-list-item-meta"
                  >
                    <div
                      className="ant-list-item-meta-avatar"
                    >
                      <span
                        className="ant-avatar ant-avatar-circle ant-avatar-image"
                        style={Object {}}
                      >
                        <img
                          onError={[Function]}
                          src="//"
                        />
                      </span>
                    </div>
                    <div
                      className="ant-list-item-meta-content"
                    >
                      <h4
                        className="ant-list-item-meta-title"
                      >
                        <a
                          href="/topic/007"
                          onClick={[Function]}
                        >
                          title
                        </a>
                      </h4>
                      <div
                        className="ant-list-item-meta-description"
                      >
                        <div>
                          <div
                            className="ant-tag ant-tag-purple"
                            style={
                              Object {
                                "backgroundColor": undefined,
                              }
                            }
                          >
                            admin
                          </div>
                          <div
                            className="ant-divider ant-divider-vertical"
                          />
                          a day ago
                        </div>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</ConnectFunction>
`;

exports[`<User /> should render default title 1`] = `
<ConnectFunction
  match={
    Object {
      "params": Object {
        "name": "admin",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className=""
  >
    <div
      className="ant-card ant-card-bordered"
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-avatar"
          >
            <span
              className="ant-avatar ant-avatar-circle ant-avatar-image"
              style={Object {}}
            >
              <img
                onError={[Function]}
                src="//"
              />
            </span>
          </div>
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              <div
                className="ant-tag ant-tag-purple"
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
              />
            </div>
            <div
              className="ant-card-meta-description"
            >
              <div>
                Github：
                <a
                  href="https://github.com/admin"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  admin
                </a>
              </div>
              <div>
                积分：
                520
              </div>
              <div>
                收藏话题：
                2
                 个
              </div>
              <div>
                注册时间：
                a day ago
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card ant-card-bordered ant-card-contain-tabs"
    >
      <div
        className="ant-card-head"
        style={Object {}}
      >
        <div
          className="ant-card-head-wrapper"
        />
        <div
          className="ant-tabs ant-tabs-top ant-card-head-tabs ant-tabs-large ant-tabs-line"
          onScroll={[Function]}
          style={Object {}}
        >
          <div
            className="ant-tabs-bar ant-tabs-top-bar ant-tabs-large-bar"
            onKeyDown={[Function]}
            role="tablist"
            style={Object {}}
            tabIndex="0"
          >
            <div
              className="ant-tabs-nav-container"
            >
              <span
                className="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
                onClick={null}
                onTransitionEnd={[Function]}
                unselectable="unselectable"
              >
                <span
                  className="ant-tabs-tab-prev-icon"
                >
                  <i
                    aria-label="icon: left"
                    className="anticon anticon-left ant-tabs-tab-prev-icon-target"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </i>
                </span>
              </span>
              <span
                className="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
                onClick={null}
                unselectable="unselectable"
              >
                <span
                  className="ant-tabs-tab-next-icon"
                >
                  <i
                    aria-label="icon: right"
                    className="anticon anticon-right ant-tabs-tab-next-icon-target"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                      />
                    </svg>
                  </i>
                </span>
              </span>
              <div
                className="ant-tabs-nav-wrap"
              >
                <div
                  className="ant-tabs-nav-scroll"
                >
                  <div
                    className="ant-tabs-nav ant-tabs-nav-animated"
                  >
                    <div>
                      <div
                        aria-disabled="false"
                        aria-selected="true"
                        className="ant-tabs-tab-active ant-tabs-tab"
                        onClick={[Function]}
                        role="tab"
                        style={
                          Object {
                            "marginRight": null,
                          }
                        }
                      >
                        最近创建的话题
                      </div>
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        className=" ant-tabs-tab"
                        onClick={[Function]}
                        role="tab"
                        style={
                          Object {
                            "marginRight": null,
                          }
                        }
                      >
                        最近参与的话题
                      </div>
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        className=" ant-tabs-tab"
                        onClick={[Function]}
                        role="tab"
                        style={
                          Object {
                            "marginRight": null,
                          }
                        }
                      >
                        收藏的话题
                      </div>
                    </div>
                    <div
                      className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            onKeyDown={[Function]}
            role="presentation"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
                "position": "absolute",
                "width": 0,
              }
            }
            tabIndex={0}
          />
          <div
            className="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
            style={
              Object {
                "marginLeft": "0%",
              }
            }
          >
            <div
              aria-hidden="false"
              className="ant-tabs-tabpane ant-tabs-tabpane-active"
              role="tabpanel"
            >
              <div
                onKeyDown={[Function]}
                role="presentation"
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": 0,
                  }
                }
                tabIndex={0}
              />
              <div
                onKeyDown={[Function]}
                role="presentation"
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "width": 0,
                  }
                }
                tabIndex={0}
              />
            </div>
            <div
              aria-hidden="true"
              className="ant-tabs-tabpane ant-tabs-tabpane-inactive"
              role="tabpanel"
            />
            <div
              aria-hidden="true"
              className="ant-tabs-tabpane ant-tabs-tabpane-inactive"
              role="tabpanel"
            />
          </div>
          <div
            onKeyDown={[Function]}
            role="presentation"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
                "position": "absolute",
                "width": 0,
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-list ant-list-split"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div
              className="ant-spin-container"
            >
              <ul
                className="ant-list-items"
              >
                <li
                  className="ant-list-item"
                >
                  <div
                    className="ant-list-item-meta"
                  >
                    <div
                      className="ant-list-item-meta-avatar"
                    >
                      <span
                        className="ant-avatar ant-avatar-circle ant-avatar-image"
                        style={Object {}}
                      >
                        <img
                          onError={[Function]}
                          src="//"
                        />
                      </span>
                    </div>
                    <div
                      className="ant-list-item-meta-content"
                    >
                      <h4
                        className="ant-list-item-meta-title"
                      >
                        <a
                          href="/topic/007"
                          onClick={[Function]}
                        >
                          title
                        </a>
                      </h4>
                      <div
                        className="ant-list-item-meta-description"
                      >
                        <div>
                          <div
                            className="ant-tag ant-tag-purple"
                            style={
                              Object {
                                "backgroundColor": undefined,
                              }
                            }
                          >
                            admin
                          </div>
                          <div
                            className="ant-divider ant-divider-vertical"
                          />
                          a day ago
                        </div>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</ConnectFunction>
`;
